sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -454,4 +454,29 @@ object DataType {
       case (fromDataType, toDataType) => fromDataType == toDataType
     }
   }
+
+  /**
+   * Compares two types, ignoring nullability of ArrayType, MapType, StructType, ignoring case
+   * sensitivity of field names in StructType as well as differences in collation for String types.
+   */
+  def equalsIgnoreCaseNullabilityAndCollation(from: DataType, to: DataType): Boolean = {
```

Comment:
I noticed that this is in `sql/api`. If we are not going to use it in the Spark Connect client,  we should just add this method to `DataTypeUtils`

Reply:
sure